6-AdvancedScenarios/1-call-api-obo/README.md

```diff
@@ -1,427 +1,361 @@
+---
+
+# Handling Conditional Access challenges in an Azure AD protected Node.js web API calling another protected Node.js web API on behalf of a user
+
+* [Overview](#overview)
+* [Scenario](#scenario)
+* [Contents](#contents)
+* [Prerequisites](#prerequisites)
+* [Setup the sample](#setup-the-sample)
+* [Explore the sample](#explore-the-sample)
+* [Troubleshooting](#troubleshooting)
+* [About the code](#about-the-code)
+* [Next Steps](#next-steps)
+* [Contributing](#contributing)
+* [Learn More](#learn-more)
 
 ## Overview
 
-This sample demonstrates a React single-page application (SPA) calling a Node.js web API (*DownstreamAPI*) protected by a **conditional access** policy via another web API (*MiddletierAPI*) using the [OAuth 2.0 on-behalf-of (OBO) flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow). The SPA project is secured using the [Microsoft Authentication Library for React](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-react) (MSAL React), while the web APIs leverage the [passport-azure-ad](https://github.com/AzureAD/passport-azure-ad) package.
+This sample demonstrates an React single-page application (SPA) which lets a user authenticate with their Azure AD tenant and obtains an [access token](https://aka.ms/access-tokens) to call an Node.js web API, protected by [passport-azure-ad](https://github.com/AzureAD/passport-azure-ad). The web API then proceeds to obtain another access token for [Microsoft Graph API](https://developer.microsoft.com/graph) using the [OAuth 2.0 on-behalf-of flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow). The web API's call to Microsoft Graph is made using the [Microsoft Graph SDK](https://docs.microsoft.com/graph/sdks/sdks-overview)
 
-The downstream web API has a [conditional access](https://docs.microsoft.com/azure/active-directory/conditional-access/overview) policy that enables [multi-factor authentication](https://docs.microsoft.com/azure/active-directory/authentication/tutorial-enable-azure-mfa) (MFA). Thus, the [Access Token](https://aka.ms/access-tokens) for the downstream web API will only be issued by Azure AD if the signed-in user has performed **MFA**. However, since the downstream web API has no user interaction capability, the [claims challenge](https://docs.microsoft.com/azure/active-directory/develop/msal-error-handling-js#conditional-access-and-claims-challenges) generated by the conditional access policy needs to be *propagated back* to the client app (SPA) via the middle-tier web API, which can then take the user back to Azure AD to get the **MFA** done.
+> :information_source: To learn how to integrate a JavaScript React application with Azure AD, consider going through the recorded session: [Deep dive on using MSAL.js to integrate React single-page applications with Azure Active Directory](https://www.youtube.com/watch?v=7oPSL5wWeS0)
 
 ## Scenario
```

Comment:
>Scenario [](http://example.com/codeflow?start=3&length=8)

No mention of handing CA challenge here?

Reply:
We are facing an issue with receiving the claims in the response from MSAL. I will update the sample once the issue is fixed.